pkg/kubectl/cmd/util/cmdproto/k8s_io_kubectl_cmd/cmd.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto2";
+package k8s.io.kubectl.cmd;
+
+import "google/protobuf/descriptor.proto";
+
+message FlagDetail {
+  required string name = 1;
+  repeated string value = 2;
+  required string shorthand = 3;
+  required string usage = 4;
+  enum EXT {
+      NONE = 0;
+      ISTIME = 1;
+      ISSTRARRAY = 2;
+      ISSTRSLICE = 3;
+  }
+  optional EXT ext = 5;
+}
+
+extend google.protobuf.FieldOptions {
+  optional FlagDetail info = 1234;
+}
+
+message Array {
+    repeated string array = 1;
+}
+
+//Flags Description
+
+
+//This is the flag description for `kubectl version`
+message VersionFlags {
+  optional bool client = 2 [ (info) = { name: "client" value: "false" shorthand: "c" usage: "Client version only (no server required)."}];
```

Comment:
You can specify the default value in a type-safe way.
https://developers.google.com/protocol-buffers/docs/proto#optional

Reply:
I can do that for all the scalar value, but default option in protos does not allow you to define a default value for aggregate fields (e.g. no defaults allowed for repeated/map fields or fields whose type is a nested message). wdyt ?